File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/placeholder/MolTagGroupPlaceholder.kt
package com.perrystreet.woof.designsystem.atomic.molecules.placeholder

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.runtime.Composable
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.AtomPlaceholder
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.roles.TagPlaceholderRole

@Composable
fun MolTagGroupPlaceholder() {
    Row(
        horizontalArrangement = Arrangement.spacedBy(SpacingRoles.Component.Compact.dp),
    ) {
        SampleTagRoles.forEach { role ->
            AtomPlaceholder(role = role)
        }
    }
}

private val SampleTagRoles = listOf(
    TagPlaceholderRole.Expanded,
    TagPlaceholderRole.Regular,
    TagPlaceholderRole.Compact,
)

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Placeholder/MolTagGroupPlaceholder.swift
import SwiftUI

public struct MolTagGroupPlaceholder: View {
    private static let sampleTagRoles: [TagPlaceholderRole] = [.expanded, .regular, .compact]

    public init() {}

    public var body: some View {
        HStack(spacing: SpacingRoles.Component.compact.rawValue) {
            ForEach(Self.sampleTagRoles, id: \.self) { role in
                AtomPlaceholder(role: role)
            }
        }
    }
}
